### PR TITLE
[dart][dart-dio] Nullable support/improvements

### PR DESCRIPTION
**Pull request by kuhnroyal**
### dart-dio/built_value

+ **Disable all fields nullable by default:**
This is not in line with OAS and will prevent future Dart nullabilty features (NNBD) from being useful as all types would be optional.

+ **Properties are `@nullable` when not required AND not nullable**

+ **Support nullable/required fields in JSON**
  * properties in built_value need to be `@nullable` when they are nullable in OAS or when they are not required in OAS
  * built_value does not support serializing `null` values by default as it is based on a serialization format based on iterables/lists and not maps
  * dart-dio uses the built_value json plugin to convert the built_value format to regular json
  * by generating a custom serializer for each class we can add support for serializing `null` values if the property is required AND nullable in OAS
  * this is a breaking change as not all properties in the models are nullable by default anymore

### Dart
* basically just serialize `null` when the field is required and nullable

This is based on https://github.com/OpenAPITools/openapi-generator/pull/8220 which needs to be merged first.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by kuhnroyal** [comment]
@josh-burton Would really like your input on this.

**Comment by kuhnroyal** [comment]
@agilob Please check it out, serialization for the normal Dart generator is a mess so I am not so sure

File: modules/openapi-generator-gradle-plugin/gradle.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # RELEASE_VERSION
-openApiGeneratorVersion=5.0.1-SNAPSHOT
+openApiGeneratorVersion=5.1.0-SNAPSHOT
+>>>>>>> origin/master
```
Comment (agilob) [inline modules/openapi-generator-gradle-plugin/gradle.properties:3]: merge conflict message here

File: modules/openapi-generator-cli/pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
         <groupId>org.openapitools</groupId>
         <artifactId>openapi-generator-project</artifactId>
         <!-- RELEASE_VERSION -->
-        <version>5.0.1-SNAPSHOT</version>
+        <version>5.1.0-SNAPSHOT</version>
```
Comment (agilob) [inline modules/openapi-generator-cli/pom.xml:7]: I think version bump should go in a separate PR, but leaving this to @wing328

File: modules/openapi-generator/src/main/resources/dart2/enum_inline.mustache [hunk 3]
```diff
@@ -1,14 +1,14 @@
 {{#description}}/// {{{description}}}{{/description}}
-class {{{classname}}}{{{enumName}}} {
+class {{{enumName}}} {
```
Comment (agilob) [inline modules/openapi-generator/src/main/resources/dart2/enum_inline.mustache:2]: I think that's a big breaking change for no big benefit, enum serialisation still won't be as good as in other languages. I think it's time to give up on doing this in openapi and use a more popular 3rd party library do it for us. I'm only familiar with json_serializable but it doesn't handle standalone enums so can't use it here. I'll look around soon.

**Comment by kuhnroyal** [comment]
Yea, wrong target and the enum branch this is based up on has already been discussed.

File: modules/openapi-generator/src/main/resources/dart2/enum_inline.mustache (hunk 3 above)
Comment (kuhnroyal) [reply modules/openapi-generator/src/main/resources/dart2/enum_inline.mustache:2]: This is actually part of  #8220 but required for this to work.

File: modules/openapi-generator-cli/pom.xml (hunk 2 above)
Comment (kuhnroyal) [reply modules/openapi-generator-cli/pom.xml]: Wrong target branch
